Assets/Src/Scripts/Input/WebCamSource.cs

```diff
@@ -0,0 +1,203 @@
+    public string DeviceName;
+    public int Fps = 30;
+    public bool Flip = false;
+    [SerializeField] private RenderTexture _buffer;
+    private WebCamTexture _webCam;
+    private int _width = 1280;
+    private int _height = 720;
+    private static readonly Dictionary<ResolutionOptions, string> _resolutionDic =
+    new Dictionary<ResolutionOptions, string>
+    {
+      { ResolutionOptions.LOW , "1280x720" } , {  ResolutionOptions.MIDDLE, "1600x900" } , { ResolutionOptions.HIGH , "1920x1080" }
+    };
+    private static readonly Dictionary<FpsOptions, string> _fpsDic =
+    new Dictionary<FpsOptions, string>
+    {
+      { FpsOptions.LOW , "30Hz" } , { FpsOptions.HiGH , "60Hz" }
+    };
+    private static readonly Dictionary<FlipOptions, string> _flipDic =
+    new Dictionary<FlipOptions, string>
+    {
+      { FlipOptions.NORMAL , "Normal" } , { FlipOptions.MIRRORING , "Mirroring" }
+    };
+    private List<string> _devicesName = new List<string>();
+    private List<string> _resolutionOpt = new List<string>();
+    private List<string> _fpsOpt = new List<string>();
+    private List<string> _flipOpt = new List<string>();
+
+    // Start button event.
+    public void Play() {
+      if (_webCam == null) {
+        _webCam = new WebCamTexture(DeviceName, _width, _height, Fps);
+        _webCam.Play();
+      }
+    }
+
+    // Stop button event.
+    public void Stop() {
+      if (_webCam != null) {
+        _webCam.Stop();
+        _webCam = null;
+      }
+    }
+
+    // DropdownChange event.
+    public void CameraDropdownChange(int value) {
+      //Debug.Log(_devicesName.ToArray()[value]);
+      if (_devicesName.ToArray()[value] != null) {
+        DeviceName = _devicesName.ToArray()[value];
+      }
+    }
+
+    // DropdownChange event.
+    public void ResolutionDropdownChange(int value) {
+      switch (value) {
+        case (int)ResolutionOptions.LOW:
+          _width = 1280;
+          _height = 720;
+          break;
+        case (int)ResolutionOptions.MIDDLE:
+          _width = 1600;
+          _height = 900;
+          break;
+        case (int)ResolutionOptions.HIGH:
+          _width = 1920;
+          _height = 1080;
+          break;
+        default:
+          break;
+      }
+    }
+
+    // DropdownChange event.
+    public void FpsDropdownChange(int value) {
+      switch (value) {
+        case (int)FpsOptions.LOW:
+          Fps = 60;
+          break;
+        case (int)FpsOptions.HiGH:
+          Fps = 30;
+          break;
+        default:
+          break;
+      }
+    }
+
+    // DropdownChange event.
+    public void FlipDropdownChange(int value) {
+      switch (value) {
+        case (int)FlipOptions.NORMAL:
+          Flip = false;
+          break;
+        case (int)FlipOptions.MIRRORING:
+          Flip = true;
+          break;
+        default:
+          break;
+      }
+    }
+
+    private void InitDropdown(Dropdown target, List<string> optarr, DropdownKey dropdownkey) {
```

Comment:
Why do you need a passed-in parameter `optarr`? It can be created or initialized inside the local scope, right?